llvm/lib/TableGen/Record.cpp

```diff
@@ -1317,13 +1308,11 @@ const Init *BinOpInit::Fold(const Record *CurRec) const {
 
       SmallVector<const Init *, 8> Args;
       SmallVector<const StringInit *, 8> ArgNames;
-      for (unsigned i = 0, e = LHSs->getNumArgs(); i != e; ++i) {
-        Args.push_back(LHSs->getArg(i));
-        ArgNames.push_back(LHSs->getArgName(i));
-      }
-      for (unsigned i = 0, e = RHSs->getNumArgs(); i != e; ++i) {
-        Args.push_back(RHSs->getArg(i));
-        ArgNames.push_back(RHSs->getArgName(i));
+      for (const DagInit *Dag : {LHSs, RHSs}) {
+        for (auto [Arg, ArgName] : Dag->getArgAndNames()) {
+          Args.push_back(Arg);
+          ArgNames.push_back(ArgName);
+        }
```

Comment:
Probably could be simplified to
```suggestion
        Args.append_range(Dag->getArgs());
        ArgNames.append_range(Dag->getArgNames());
```

Reply:
Done. I modified the `getArgAndNames` function to return a range of std::pair, so I can use just once append_range() to build the args for the new Dag being created.